cfgov/ask_cfpb/models/search/documents.py

```diff
@@ -0,0 +1,508 @@
+        test = Truncator(text).words(word_limit, truncate=' ...')
+        if len(test) <= 255:
+            return test
+        else:
+            word_limit -= 1
+
+
+def extract_raw_text(stream_data):
+    # Extract text from stream_data, starting with the answer text.
+    text_chunks = [
+        block.get('value').get('content')
+        for block in stream_data
+        if block.get('type') == 'text'
+    ]
+    extra_chunks = [
+        block.get('value').get('content')
+        for block in stream_data
+        if block.get('type') in ['tip', 'table']
+    ]
+    chunks = text_chunks + extra_chunks
+    return " ".join(chunks)
+
+
+label_autocomplete = analyzer(
+    'label_autocomplete',
+    tokenizer=tokenizer(
+        'trigram',
+        'edge_ngram',
+        min_gram=2,
+        max_gram=25,
+        token_chars=["letter", "digit"]
+    ),
+    filter=['lowercase', token_filter('ascii_fold', 'asciifolding')]
+)
+
+synonynm_filter = token_filter(
+    'synonym_filter_en',
+    'synonym',
+    synonyms_path='/usr/share/elasticsearch/config/synonyms/synonyms_en.txt'
+)
+
+synonym_analyzer = analyzer(
+    'synonym_analyzer_en',
+    type='custom',
+    tokenizer='standard',
+    filter=[
+        synonynm_filter,
+        'lowercase'
+    ])
+
+
+def get_ask_breadcrumbs(language='en', portal_topic=None):
+    default_crumbs = {
+        'es': [{
+            'title': 'Obtener respuestas', 'href': '/es/obtener-respuestas/',
+        }],
+        'en': [{
+            'title': 'Ask CFPB', 'href': '/ask-cfpb/',
+        }],
+    }
+    if portal_topic:
+        page = get_portal_or_portal_search_page(
+            portal_topic=portal_topic, language=language)
+        crumbs = [{
+            'title': page.title,
+            'href': page.url
+        }]
+        return crumbs
+    return default_crumbs[language]
+
+
+def get_portal_or_portal_search_page(portal_topic, language='en'):
+    if portal_topic:
+        portal_page = portal_topic.portal_pages.filter(
+            language=language, live=True).first()
+        if portal_page:
+            return portal_page
+        else:
+            portal_search_page = portal_topic.portal_search_pages.filter(
+                language=language, live=True).first()
+            return portal_search_page
+    return None
+
+
+REUSABLE_TEXT_TITLES = {
+    'about_us': {
+        'en': 'About us (For consumers)',
+        'es': 'About us (for consumers) (in Spanish)'
+    },
+    'disclaimer': {
+        'en': 'Legal disclaimer for consumer materials',
+        'es': 'Legal disclaimer for consumer materials (in Spanish)'
+    }
+}
+
+
+def get_reusable_text_snippet(snippet_title):
+    try:
+        return ReusableText.objects.get(
+            title=snippet_title)
+    except ReusableText.DoesNotExist:
+        pass
+
+
+def get_standard_text(language, text_type):
+    return get_reusable_text_snippet(
+        REUSABLE_TEXT_TITLES[text_type][language]
+    )
+
+
+class AnswerPage(CFGOVPage):
```

Comment:
@alexm118 does the `AnswerPage` model need to live with the search document, or does it just need to live in a separate model file in `ask_cfpb.models`?

Reply:
Yeah, I doesn't need to live within documents.py it just cannot be in `pages.py` or `views.py`. 

I didn't think about it at the time, but potentially we may want to actually move the `PortalSearchPage` out of pages.py into something that more aligns with what its doing? Just a thought, as it seeems to be the offending class that actually has more logic than a standard wagtail page embedded within it.

Reply:
That'd be fine too — there's nothing that says all the Wagtail pages have to live in a `models/pages.py`. I think that's a convention we accidentally fell into.

I actually think the fact that we *don't* split up our page models more is part of how we end up with circular dependencies.

Reply:
Moved AnswerPage class from search documents into ask_cfpb.models.

Reply:
@alexm118 Perhaps moving `PortalSearchPage` out of pages.py could be done in a different PR.